### PR TITLE
Remove excess memory allocation

### PR DESCRIPTION
**Pull request by jcready**

### CODE REVIEW

**Comment by jaxgeller** [comment]
Thanks for the PR, will give this a look tomorrow

File: src/tweezer.js [hunk 1]
```diff
@@ -29,43 +30,44 @@ export default class Tweezer {
...
+    const lastTick = this.next === null ? this.start : this.next // this.next can be 0
 
     if (!this.timeStart) this.timeStart = currentTime
     this.timeElapsed = currentTime - this.timeStart
     this.next = Math.round(this.ease(this.timeElapsed, this.start, this.end - this.start, this.duration))
 
     if (this._shouldTick(lastTick)) {
-      this.emit('tick', this.next)
-      this.frame = requestAnimationFrame(this._tick.bind(this))
+      this.emit('tick', this.next).frame = requestAnimationFrame(this._tick)
```
Comment (Jacse) [inline src/tweezer.js:54]: I agreee with the PR over all, but don't these two simplifications make the code less readable?

**Comment by jaxgeller** [comment]
Cool so first off thanks for the pr, just finished testing this in the major browsers and running perf tests. So  I agree with mostly everything in this pr but theres some stylistic changes I'd like to see happen before I merge this in, which I'll make those comments inline with the pr.

File: src/tweezer.js [hunk 2]
```diff
@@ -29,43 +30,44 @@ export default class Tweezer {
...
+    else this.events[name] = [handler]
     return this
   }
 
   emit (name, val) {
-    let e = this.events[name]
-    e && e.forEach(handler => handler.call(this, val))
+    const e = this.events[name]
+    if (e) e.forEach(handler => handler.call(this, val))
+    return this
```
Comment (jaxgeller) [inline src/tweezer.js:42]: Emit is really more of a private method for the class and I'd like to keep it that way.

File: src/tweezer.js (hunk 1 above)
Comment (jaxgeller) [inline src/tweezer.js:54]: Agree here with @Jacse I'd prefer to keep code quality over this micro optimization.

File: src/tweezer.js [hunk 3]
```diff
@@ -29,43 +30,44 @@ export default class Tweezer {
...
   }
+}
 
-  _shouldTick (lastTick) {
-    return {
-      up: this.next < this.end && lastTick <= this.next,
-      down: this.next > this.end && lastTick >= this.next
-    }[this.direction]
-  }
+function defaultEase (t, b, c, d) {
```
Comment (jaxgeller) [inline src/tweezer.js:61]: I'd prefer these next three functions to be within the class, but out of the three I can justify this being outside of the class.

File: src/tweezer.js [hunk 4]
```diff
@@ -29,43 +30,44 @@ export default class Tweezer {
...
+  return (t /= d / 2) < 1
+    ? c / 2 * t * t + b
+    : -c / 2 * ((--t) * (t - 2) - 1) + b
+}
 
-  _defaultEase (t, b, c, d) {
-    if ((t /= d / 2) < 1) return c / 2 * t * t + b
-    return -c / 2 * ((--t) * (t - 2) - 1) + b
-  }
+function shouldTickUp (lastTick) {
```
Comment (jaxgeller) [inline src/tweezer.js:67]: For these next two functions, they should be in the class as private methods, not a fan of the function inlining.

Comment (jcready) [inline src/tweezer.js:67]: These functions are more private now than they would be inside the class. I could wrap these functions plus the class in a closure and only return the class if you're worried about this somehow getting used in the browser without a build step and having them show up as globals.

Comment (jaxgeller) [inline src/tweezer.js:67]: Sorry for the delay. I'd just like to keep these functions within the class itself to avoid confusion.

**Comment by jonahbron** [comment]
@jcready based on some of the review feedback here, I'm going to close this PR for now.  If you get a chance to make some of the changes described, please re-open it.  Thank you 😃

**Pull request closed without merging**